### PR TITLE
feat: upgrade to vee-validate v3

### PR DESCRIPTION
**Pull request by devCrossNet**
### What is accomplished by your PR?
This PR upgrates all components, tests and storybook to vee-validate version 3.x

### Is there something controversial in your PR?
I needed to add a new dependency `flush-promises` in order to make the tests working

### Link to the Issue
- via slack

# Checklist

### New Feature / Bug Fix

- [x] Run unit tests to ensure all existing tests are still passing
- [x] Add new passing unit tests to cover the code introduced by your PR
- [ ] Change documentation for the code introduced by your PR
- [ ] Add Story / Design System Page for a new component introduced by your PR

### CODE REVIEW

File: src/app/app.ts [hunk 1]
```diff
@@ -1,18 +1,23 @@
...
 import { store } from './store';
 import { router } from './router';
 import { IState } from './state';
 import { i18n } from '@shared/plugins/i18n/i18n';
 import { HttpService } from '@shared/services/HttpService/HttpService';
+import App from './app/App/App.vue';
 import './shared/directives';
 
-Vue.use(VeeValidate, { inject: false, delay: 1 });
+const { required, email, integer, min } = require('vee-validate/dist/rules.umd.js');
```
Comment (xMartin) [inline src/app/app.ts:15]: Consider adding a comment why this inconsistent import is necessary.

Comment (devCrossNet) [reply src/app/app.ts:16]: Done 👍

**Review by xMartin-dv — CHANGES_REQUESTED** [review]
When I open the form example page, the dropdown for country is initially invalid.

**Comment by xMartin** [comment]
Memory consumption looks good now!

This change was requested by me in the community Slack:
> I’m running into a memory leak with vee-validate. It got fixed in version 2.1 (https://github.com/logaretm/vee-validate/releases/tag/2.1.0-beta.2) so I think we should update. Unfortunately there are breaking changes in 2.1 and without any code changes all form element are initially invalid  for me.
As the “next” branch of vuesion is using vee-validate v3, I was wondering how much effort it would be  to update it on the stable vuesion or if there’s any ideas why what the validation issue with v2.1 could be.

**Comment by devCrossNet** [comment]
@xMartin fixed, there is still an issue with validating the checkbox but I also have a workaround. I pushed it and you can comment if this would be enough or if we should further investigate

File: src/app/shared/components/VueSelect/VueSelect.vue [hunk 2]
```diff
@@ -1,32 +1,35 @@
...
-      <option v-for="(option, idx) in selectOptions" :key="idx" :value="option.value" :selected="isSelected(option)">
-        {{ option.label }}
-      </option>
-    </select>
-    <i :class="$style.icon" v-if="!multiple"></i>
-    <span :class="$style.bar"></span>
-    <label :for="id">{{ placeholder }}<sup v-if="required">*</sup></label>
-  </div>
+  <ValidationProvider v-slot="{ errors }" ref="validator" :vid="id" :name="name" :rules="validation">
+    {{ errors }} {{ value }}
```
Comment (xMartin) [inline src/app/shared/components/VueSelect/VueSelect.vue:3]: This seems to be left over from debugging, right?

**Review by xMartin-dv — COMMENTED** [review]
Dropdown works correctly now.

File: src/app/example/form/FormExample/FormExample.vue [hunk 3]
```diff
@@ -1,134 +1,135 @@
...
 
-    <vue-grid-row>
-      <vue-grid-item>
-        <vue-checkbox
-          name="acceptTerms"
-          id="acceptTerms"
-          v-model="form.acceptTerms"
-          label="I accept the terms"
-          validation="required"
-          required
```
Comment (xMartin) [inline src/app/example/form/FormExample/FormExample.vue:102]: Why don't we validate this checkbox? Seems to be very unintuitive having to check this one separately in the button.

File: src/app/shared/components/VueSelect/VueSelect.vue (hunk 2 above)
Comment (devCrossNet) [reply src/app/shared/components/VueSelect/VueSelect.vue:3]: upps, yes

File: src/app/example/form/FormExample/FormExample.vue (hunk 3 above)
Comment (devCrossNet) [reply src/app/example/form/FormExample/FormExample.vue:102]: I didn't find a way that the validation works correctly, I need to investigate more. The documentation says that "false" should work as "not required" but somehow it doesn't pick it up

**Pull request closed without merging**